packages/trader/src/AppV2/Components/MarketSelector/market-selector.tsx

```diff
@@ -26,9 +31,20 @@ const MarketSelector = observer(() => {
     if (typeof currentSymbol?.exchange_is_open === 'undefined')
         return <Skeleton.Square height={42} width={240} rounded />;
 
+    const onClick = () => {
+        if (guide_dtrader_v2?.market_selector) {
+            setGuideState('should_run_market_selector_guide', false);
+        } else {
+            const latest_guide_dtrader_v2 = getLocalStorage('guide_dtrader_v2');
```

Comment:
`useLocalStorageData` hook is using `getLocalStorage` function under the hood. Taking it into consideration, can we use `guide_dtrader_v2` (from `useLocalStorageData`) directly?

Reply:
Even setData(prev => {...prev, new_flag:true})? Okay

Reply:
The values inside the hook are not updated, it is not properly implemented in the first place, so I made sure it always gets the latest values from store. Before that it was just overriding each other with outdated values.